cns/grpc/proto/server.proto

```diff
@@ -0,0 +1,44 @@
+syntax = "proto3";
+
+package cns;
+
+// The Container Network Service (CNS) exposes a set of operations that allow the Distributed Network Controller (DNC) to manage
+// and monitor nodes in an orchestrator's infrastructure.
+
+// CNSService defines the gRPC service exposed by CNS to interact with DNC.
+service CNSService {
+  // Sets the orchestrator information for a node.
+  rpc SetOrchestratorInfo(SetOrchestratorInfoRequest) returns (SetOrchestratorInfoResponse);
+
+  // Retrieves detailed information about a specific node.
+  // Primarily used for health checks.
+  rpc GetNodeInfo(NodeInfoRequest) returns (NodeInfoResponse);
+}
+
+// SetOrchestratorInfoRequest is the request message for setting the orchestrator information.
+message SetOrchestratorInfoRequest {
+  string dncPartitionKey = 1; // The partition key for DNC.
+  string nodeID = 2; // The node ID.
+  string orchestratorType = 3; // The type of the orchestrator (e.g., ServiceFabric, Kubernetes, etc.).
+}
+
+// SetOrchestratorInfoResponse is the response message for setting the orchestrator information.
+message SetOrchestratorInfoResponse {
+  int32 returnCode = 1; // The return code of the operation.
```

Comment:
gRPC is different from HTTP, so it has different mechanisms for communicating whether a RPC call failed with an error. There are built-in ways of communicating error codes and messages through the error that is returned by a gRPC operation in Go code if the call fails (https://grpc.io/docs/guides/status-codes/). We can remove `returnCode` and `message` here